module/modmanager/module.go

```diff
@@ -371,14 +377,14 @@ func (m *module) cleanupAfterStartupFailure() {
 }
 
 func (m *module) cleanupAfterCrash(mgr *Manager) {
-	utils.UncheckedError(m.sharedConn.Close())
-	mgr.rMap.Range(func(r resource.Name, mod *module) bool {
-		if mod == m {
-			mgr.rMap.Delete(r)
-		}
-		return true
-	})
-	mgr.modules.Delete(m.cfg.Name)
+	m.deregisterResources()
+	if err := m.sharedConn.Close(); err != nil {
+		m.logger.Warnw("Error closing connection to crashed module", "error", err)
```

Comment:
[nit] I think this could still be an unchecked error, but seems fine.

Reply:
This was copied from the old version of the OUE handler. It kind of is unchecked: all we do with `err` is log it and continue with the rest of the cleanup steps. I'm inclined to leave it that way unless we think it will cause too much log noise.

Reply:
> Are you saying we should do that rather than log via the module scoped logger here?

Whoops, no. I also forgot what `utils.UncheckedError` here means. I meant that we should completely ignore the error and not log it at all. It's not a super strong opinion, nor do I think totally ignoring the error is critical. This is just a pattern NetCode comes up against often: a failure to close something, we've found, is rarely an interesting thing to report to a user, especially at the `Error` log level.